src/fsharp/CompileOps.fs

```diff
@@ -1639,15 +1639,111 @@ let GetDefaultFSharpCoreReference() = typeof<list<int>>.Assembly.Location
 
 // If necessary assume a reference to the latest System.ValueTuple with which those tools are built.
 let GetDefaultSystemValueTupleReference() = 
-    try 
-       let asm = typeof<System.ValueTuple<int,int>>.Assembly
-       if asm.FullName.StartsWith "System.ValueTuple" then 
-           Some asm.Location 
-       else None
-    with _ -> None
+    let path = Path.GetDirectoryName(GetDefaultFSharpCoreReference())
```

Comment:
OK, yes, this is wrong when running the compiler or F# Interactive in  .NET Core without --noframework etc.